src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -2524,4 +2534,29 @@ void SurgeGUIEditor::showSettingsMenu(CRect &menuRect)
 
 }
 
+CPoint SurgeGUIEditor::getCorrectlyScaledMouseLocation()
+{
+    /*
+    ** Please see github issue 356 for discussion on this.
+    **
```

Comment:
You could document properly what this function does with javadoc style in the class definition.

Reply:
I think we could document what it does in the class definition; but should document how it does it here.

What it does is get the correctly scaled mouse location. I've always been skeptical of

```
  /**
   * getCorrectlyScaledMouseLocation
   * @decscription gets the correctly scaled mouse location, overcoming some vstgui bugs if needed
   * @returns the correctly scaled mouse location
   */
   CPoint getCorrectlyScaledMouseLocation()
```

as being useful (which is why i generally think the oxygen comments are tough for protected and internal methods which are well named).

So I guess: how would you separate out the comment between intent and implementation?

Reply:
The correct format would be:

```
 /**
  * <short summary>
  * @return <what gives back>
  *
  * <long description what this does and maybe some implementation details>
  */
 CPoint getCorrectlyScaledMouseLocation()
```

In the long description you would explain what you mean by correctly scaled mouse location. It is far from obvious.

Reply:
Maybe the function name is not good because correctly, in fact, does not mean anything. Dunno.

Reply:
I changed the function name to

```
   /**
    * getCurrentMouseLocationCorrectedForVSTGUIBugs
    *
    * This function gets the current mouse location for the frame
    * but adds, in necessary cases, workarounds for bugs in the
    * vstgui framework. Use it in place of frame->getCurrentMouseLocation
    */
   CPoint getCurrentMouseLocationCorrectedForVSTGUIBugs();
```

since I think that is the most correct name (since it replaces getCurrentMouseLocation)